RecoHGCal/TICL/plugins/TrackstersMergeProducer.cc

```diff
@@ -0,0 +1,565 @@
+              << t.eigenvectors[0].eta() << ", " << t.eigenvectors[0].phi() << std::endl;
+          diff_pt -= gamma_pt;
+          if (diff_pt > 2.) {
+            // Create also a neutral on top, with calo direction and diff_pt as energy
+            LogDebug("TrackstersMergeProducer")
+                << " Adding also a neutral hadron from TRK Trackster with energy " << diff_pt << " and direction "
+                << t.eigenvectors[0].eta() << ", " << t.eigenvectors[0].phi() << std::endl;
+          }
+        } else {
+          // Create neutral with calo direction
+          LogDebug("TrackstersMergeProducer")
+              << " Creating a neutral hadron from TRK Trackster with energy " << diff_pt << " and direction "
+              << t.eigenvectors[0].eta() << ", " << t.eigenvectors[0].phi() << std::endl;
+        }
+      }
+      // If the energy is in the correct ball-park (this also covers the
+      // previous case after the neutral emission), use weighted averages
+      if (diff_pt_sigmas > -pt_sigma_low_) {
+        // Create either an electron or a charged hadron, using the weighted
+        // average information between the track and the cluster for the
+        // energy, the direction of the track at the vertex.  The PID is simply
+        // passed over to the final trackster, while the energy is changed.
+        auto average_pt = (w_cal * t.raw_pt + trk_pt * w_trk) / (w_cal + w_trk);
+        LogDebug("TrackstersMergeProducer")
+            << " Creating electron/charged hadron from TRK Trackster with weighted p_t " << average_pt
+            << " and direction " << track.eta() << ", " << track.phi() << std::endl;
+
+      }
+      // If the energy of the calo is too low, just use track-only information
+      else {
+        // Create either an electron or a charged hadron, using the track
+        // information only.
+        LogDebug("TrackstersMergeProducer")
+            << " Creating electron/charged hadron from TRK Trackster with track p_t " << trk_pt << " and direction "
+            << track.eta() << ", " << track.phi() << std::endl;
+      }
+      result->push_back(t);
+      usedTrackstersTRK[tracksterTRK_idx] = 1;
+      tracksterTRK_idx++;
+    }
+  }
+
+  tracksterTRK_idx = 0;
+  for (auto const &t : trackstersTRK) {
+    if (debug_) {
+      LogDebug("TrackstersMergeProducer") << " Considering trackster " << tracksterTRK_idx
+                                          << " as used: " << usedTrackstersTRK[tracksterTRK_idx] << std::endl;
+    }
+    if (!usedTrackstersTRK[tracksterTRK_idx]) {
+      LogDebug("TrackstersMergeProducer")
+          << " Creating a charge hadron from TRK Trackster with track energy " << t.raw_energy << " and direction "
+          << t.eigenvectors[0].eta() << ", " << t.eigenvectors[0].phi() << std::endl;
+      result->push_back(t);
+    }
+    tracksterTRK_idx++;
+  }
+
+  auto tracksterEM_idx = 0;
+  for (auto const &t : trackstersEM) {
+    if (debug_) {
```

Comment:
Even here, the usage of the ```LogDebug``` and the ```debug_``` flag at the same time is rather peculiar.
Here however it could be justified because, once you switch on the LogDebug, then you can print additional  infos when ```debug_ = true```

Reply:
Please, read my reply at the previous comment.